test/GPv2Settlement/Settle/Reentrancy.t.sol

```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+pragma solidity ^0.8.26;
+
+import {IERC20, GPv2Order, GPv2Interaction, GPv2Settlement} from "src/contracts/GPv2Settlement.sol";
+
+import {Harness, Helper} from "../Helper.sol";
+
+import {Settlement} from "test/libraries/Settlement.sol";
+import {Sign} from "test/libraries/Sign.sol";
+import {Trade} from "test/libraries/Trade.sol";
+import {Order} from "test/libraries/Order.sol";
+
+import {SwapEncoder} from "test/libraries/encoders/SwapEncoder.sol";
+import {SettlementEncoder} from "test/libraries/encoders/SettlementEncoder.sol";
+
+// solhint-disable func-name-mixedcase
+contract Reentrancy is Helper {
+    using Settlement for Harness;
+    using Trade for GPv2Order.Data;
+    using SettlementEncoder for SettlementEncoder.State;
+    using SwapEncoder for SwapEncoder.State;
+
+    function test_revert_rejects_reentrancy_attempts_via_interactions() public {
+        reject_reentrancy_attempts_via_interactions(owner, settle_reentrancy_calldata());
+        reject_reentrancy_attempts_via_interactions(owner, swap_reentrancy_calldata());
+    }
+
+    function test_revert_rejects_reentrancy_attempts_even_as_a_registered_solver() public {
+        reject_reentrancy_attempts_via_interactions(address(settlement), settle_reentrancy_calldata());
+        reject_reentrancy_attempts_via_interactions(address(settlement), swap_reentrancy_calldata());
+    }
+
+    function reject_reentrancy_attempts_via_interactions(address who, bytes memory data) internal {
```

Comment:
I believe the signature should rather be:
```suggestion
    function reject_reentrancy_attempts_via_interactions(bytes memory data, bool settlementContractIsSolver) internal {
```
There's no specific need to call the first `settle` with a specific address, any solver would work for this and can be the same in both cases.
If the new flag is true the settlement contract is made a solver, otherwise nothing happens.